darts/metrics/metrics.py

```diff
@@ -872,9 +933,101 @@ def r2_score(actual_series: Union[TimeSeries, Sequence[TimeSeries]],
+    reduction
+        Function taking as input a `np.ndarray` and returning a scalar value. This function is used to aggregate
+        the metrics of different components in case of multivariate `TimeSeries` instances.
+    inter_reduction
+        Function taking as input a `np.ndarray` and returning either a scalar value or a `np.ndarray`.
+        This function can be used to aggregate the metrics of different series in case the metric is evaluated on a
+        `Sequence[TimeSeries]`. Defaults to the identity function, which returns the pairwise metrics for each pair
+        of `TimeSeries` received in input. Example: `inter_reduction=np.mean`, will return the average of the pairwise
+        metrics.
+    n_jobs
+        The number of jobs to run in parallel. Parallel jobs are created only when a `Sequence[TimeSeries]` is
+        passed as input, parallelising operations regarding different `TimeSeries`. Defaults to `1`
+        (sequential). Setting the parameter to `-1` means using all the available processors.
+    verbose
+        Optionally, whether to print operations progress
+    Returns
+    -------
+    float
+        The Mean Percentage Likelihood
+    """
+
+    # Can't use _get_values since we are interested in the samples as well
+    raise_if_not(actual_series.width == pred_series.width, "The two time series must have the same number of components",
+                 logger)
+
+    raise_if_not(isinstance(intersect, bool), "The intersect parameter must be a bool")
+
+    actual_series_common = actual_series.slice_intersect(pred_series) if intersect else actual_series
+    pred_series_common = pred_series.slice_intersect(actual_series) if intersect else pred_series
+
+    raise_if_not(actual_series_common.has_same_time_as(pred_series_common), 'The two time series (or their intersection)'
+                                                                            'must have the same time index.'
+                                                                            '\nFirst series: {}\nSecond series: {}'.format(
+                                                                            actual_series.time_index, pred_series.time_index), logger)
+
+    raise_if_not(actual_series.is_deterministic and pred_series.is_stochastic,
+                 "Expected actual_series (ground truth) to be deterministic"
+                 "and pred_series to be a probabilistic forecast")
+
+    values = actual_series_common.univariate_values()
+    samples = pred_series_common.all_values()[:, 0, :]
+
+    # Remove nans
+    isnan_mask = np.logical_or(np.isnan(values), np.any(np.isnan(samples), axis=1))
+    values = np.delete(values, isnan_mask)
+    samples = np.delete(samples, isnan_mask, axis=0)
+
+    # todo vectorize, gaussian_kde cannot be called on all the samples directly
+    n = len(values)
+    mean_probability = 0.0
+    for i in range(n):
+        value = values[i]
+        kde = gaussian_kde(samples[i, :])
```

Comment:
I don't think you want to use a kernel estimator here. It probably doesn't do what you want.
Instead, you should probably use the simple normal distribution functions: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.norm.html

Something like:
```
mu, sigma = np.mean(samples[i, :]), np.std(samples[i, :])

# for estimating a probability
proba = cdf(value + deviation, loc=mu, scale=sigma) - cdf(value - deviation, loc=mu, scale=sigma)

# for estimating a density (or likelihood)
likelihood = pdf(value, loc=mu, scale=sigma)

# we should probably leave it as an option to return the log-likelihood instead:
log_likelihood = logpdf(value, loc=mu, scale=sigma)
```